source/Generic/ExtraMetadataToolsVideo/ExtraMetadataToolsVideo.psm1

```diff
@@ -1049,29 +1266,74 @@ function Set-YouTubeVideoManual
     }
 
     $game = $gameDatabase[0]
+    $search = '"' + "https://www.youtube.com/watch?v=" + $YouTubeID + '"'
+
+    $success = Download-YoutubeVideo $game $search $true
+    if($success) 
+    {
+        $PlayniteApi.Dialogs.ShowMessage(([Playnite.SDK.ResourceProvider]::GetString("LOCExtra_Metadata_tools_GenericFinishedMessage")), "Extra Metadata tools")
+    } 
+    else 
+    {
+        $PlayniteApi.Dialogs.ShowMessage(([Playnite.SDK.ResourceProvider]::GetString("LOCExtra_Metadata_tools_GenericErrorProcessingFileMessage")), "Extra Metadata tools")
+    }
+}
+
+function Download-YoutubeVideo
+{
+    param (
+        $game,
+        [string]$search,
+        [bool]$overwrite
+    )
 
     $extraMetadataDirectory = Set-GameDirectory $game
-    $videoPath = Join-Path $extraMetadataDirectory -ChildPath "VideoTrailer.mp4"
-    $videoTempPath = Join-Path $extraMetadataDirectory -ChildPath "VideoTemp.mp4"
-    $youtubedl = $settings.youtubedlPath
-    $search = '"' + "https://www.youtube.com/watch?v=" + $YouTubeID + '"'
-    
+
+    $ytdlp = $settings.ytdlpPath
+    [string]$ytdlpSettings = Get-YTDLPSettings
+
+    $videoPathNoExtension = Join-Path $extraMetadataDirectory -ChildPath "VideoTrailer."
+    $videoTempPathNoExtension = Join-Path $extraMetadataDirectory -ChildPath "VideoTemp."
+
+
+    $filePresent = ((Get-ChildItem -Path "$videoPathNoExtension*").Count -ne 0)
```

Comment:
I don't fully understand why it's using GCI here, can you explain please?

ExtraMetadataTools video loading only seeks `VideoTrailer.mp4` inside the game EMT directory.

For that reason, it should be using the `Test-Path` cmdlet or the `Exists` method from the net File class

Reply:
You are right . That is a leftover from checking if .webm without converting would be ok.
Done in 51bbcf88bcea825a6ffb414ed668b2ac5ec39ad1